S23/fulvius31/app/src/main/java/lowercase/Lowercaser.java

```diff
@@ -0,0 +1,115 @@
+    private char[] strChar;
+
+    public Lowercaser(String upperCase, String lang) {
+        this.upperCase = upperCase;
+        this.lang = lang.split("-")[0];
+    }
+
+    private String lowerTrAz() {
+        for (int i = 0; i < strChar.length; i++) {
+            if (strChar[i] == 'I')
+                strChar[i] = 'ı';
+            else
+                strChar[i] = Character.toLowerCase(strChar[i]);
+        }
+        return new String(strChar);
+    }
+
+    private String lowerGa() {
+        Set<String> irishVowels = new HashSet<>
+                (Arrays.asList("A", "E", "I", "O", "U", "Á", "É", "Í", "Ó", "Ú"));
+        String normalizedUpper = Normalizer.normalize(upperCase,
+                Normalizer.Form.NFC);
+
+        StringBuilder sb = new StringBuilder();
+        boolean isSpecialCase = false;
+        for (int i = 0; i < normalizedUpper.length(); i++) {
+            String ch = normalizedUpper.substring(i, i + 1);
+            if (i == 0 && (ch.contains("n") || ch.contains("t"))) {
+                isSpecialCase = true;
+            }
+            if (i == 1 && irishVowels.contains(ch) && isSpecialCase) {
+                sb.append("-");
+            }
+            sb.append(ch.toLowerCase());
+        }
+        return sb.toString();
+    }
+
+    private String lowerEl() {
+        int strCharLength = strChar.length;
+        if (strChar[strCharLength - 1] == 'Σ') {
+            strChar[strCharLength - 1] = 'ς';
+            strCharLength--;
+        }
+        for (int i = 0; i < strCharLength; i++) {
+            strChar[i] = Character.toLowerCase(strChar[i]);
+        }
+        return new String(strChar);
+    }
+
+    public String getLowerCase() {
+        if (isUnchangedLang())
+            return upperCase;
+        if (!isExceptionLang())
+            return upperCase.toLowerCase();
+        strChar = upperCase.toCharArray();
+        if (lang.equals(LanguagesToExamine.AZERBAIJANI.getAcronym()) ||
+                lang.equals(LanguagesToExamine.TURKISH.getAcronym())) {
+            return lowerTrAz();
+        }
+        if (lang.equals(LanguagesToExamine.IRISH.getAcronym())) {
+            return lowerGa();
+        }
+        if (lang.equals(LanguagesToExamine.GREEK.getAcronym())) {
+            return lowerEl();
+        }
+
+        return upperCase;
+    }
+
+    private boolean isUnchangedLang() {
+        return lang.equals(LanguagesToExamine.CHINESE.getAcronym()) ||
+                lang.equals(LanguagesToExamine.JAPANESE.getAcronym()) ||
+                lang.equals(LanguagesToExamine.THAI.getAcronym());
+    }
+
+    private boolean isExceptionLang() {
+        return lang.equals(LanguagesToExamine.TURKISH.getAcronym()) ||
+                lang.equals(LanguagesToExamine.AZERBAIJANI.getAcronym()) ||
+                lang.equals(LanguagesToExamine.IRISH.getAcronym()) ||
+                lang.equals(LanguagesToExamine.GREEK.getAcronym());
+    }
+
+    enum LanguagesToExamine {
```

Comment:
Maybe we can use any function that would work for any language instead of dynamically giving some languages to the code (Locale)

Reply:
Thanks for your review!

Could you provide me some examples and why it would be better than use an enum ?

Reply:
by enum data structure you have only hard coded few of the languages but dynamically we can't use enum. It would be better we we give languages dynamically

Reply:
thanks for your comment on it. 

enum is not hard coded, enum is a data type. It is used - in this case - also to improve the readability and the maintainability of the code. For example, if you have to add "it" language, maybe the reader does not know that "it" is for the italian language. So who writes the code, knows it and put it in an enum as ITALIAN with its acronym "it".

The purpose of the function was not to make the application interactive with the users but to test it thru JUnit tests (in this case).

Tell me if you need more details.